### PR TITLE
Add @yoast/search-metadata-previews package

### PR DESCRIPTION
**Pull request by diedexx**
Fixes https://github.com/Yoast/javascript/issues/31

Moved the following to a new package (`@yaost/search-metadata-previews`)
- `yoast-components/composites/SnippetEditor`
- `yoast-components/composites/SnippetPreview`

There still are some references to yoast-components, which will trigger an ECYCLE dependency cycle warning. I think we can ignore these until we've moved all code from `yoast-components/composites/plugin/shared` to another package or find another solution.

## Test instructions
- run `yarn test`. This should succeed. 
- run `yarn lint`. This should succeed.

### CODE REVIEW

File: apps/components/SnippetEditorExample.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 // External dependencies.
 import React, { Component } from "react";
 
+// Internal dependencies.
 import ExamplesContainer from "./ExamplesContainer";
 import { assessments } from "yoastseo";
-const { MetaDescriptionLengthAssessment } = assessments.seo;
+import { SnippetEditor } from "@yoast/search-metadata-previews";
```
Comment (bintzandt) [inline apps/components/SnippetEditorExample.js]: This is not really an internal dependency 🤷‍♂️

File: packages/search-metadata-previews/SnippetEditor/components/ModeSwitcher.js [hunk 2]
```diff
@@ -2,14 +2,14 @@
 import React from "react";
 import styled from "styled-components";
 import { __ } from "@wordpress/i18n";
+import PropTypes from "prop-types";
 
 // Internal dependencies.
-import { Button } from "../../Shared/components/Button";
+import { Button } from "yoast-components/composites/Plugin/Shared/components/Button";
+import { SvgIcon } from "@yoast/components";
```
Comment (bintzandt) [inline packages/search-metadata-previews/SnippetEditor/components/ModeSwitcher.js]: Same comment as before, I don't think that these import are really internal.

File: packages/search-metadata-previews/SnippetEditor/components/SnippetEditor.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ const { MetaDescriptionLengthAssessment, PageTitleWidthAssessment } = assessment
 import { helpers } from "yoastseo";
 const { measureTextWidth } = helpers;
 
-import stripSpaces from "yoastseo/src/stringProcessing/stripSpaces";
+import { string } from "yoastseo";
```
Comment (bintzandt) [inline packages/search-metadata-previews/SnippetEditor/components/SnippetEditor.js]: I like this change :)

File: packages/search-metadata-previews/SnippetPreview/components/HelpTextWrapper.js [hunk 4]
```diff
@@ -5,11 +5,11 @@ import styled from "styled-components";
 import uniqueId from "lodash/uniqueId";
 
 // Internal dependencies
-import HelpText from "../../Shared/components/HelpText";
-import { colors, rgba } from "@yoast/components/style-guide";
-import { Button } from "../../Shared/components/Button";
+import HelpText from "yoast-components/composites/Plugin/Shared/components/HelpText";
```
Comment (bintzandt) [inline packages/search-metadata-previews/SnippetPreview/components/HelpTextWrapper.js]: I'm guessing that this reference needs to change after the `Plugin/Shared/components` are merged into `@yoast/components`.

File: packages/search-metadata-previews/index.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export * from "./SnippetEditor";
```
Comment (bintzandt) [inline packages/search-metadata-previews/index.js]: I am not sure, but shouldn't we assign names to these exports?

File: packages/search-metadata-previews/jest/setupTests.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import "jest-styled-components";
+import Enzyme from "enzyme";
+import EnzymeAdapter from "enzyme-adapter-react-16";
```
Comment (bintzandt) [inline packages/search-metadata-previews/jest/setupTests.js]: Do we still need `Enzyme` here? I found that my tests were also working when I removed enzyme.

File: packages/yoast-components/index.js [hunk 7]
```diff
@@ -66,16 +66,15 @@ export {
...
 export * from "./forms";
 export * from "./composites/Plugin/ContentAnalysis";
+export * from "@yoast/search-metadata-previews";
 export { default as utils } from "./utils";
 export { localize } from "./utils/i18n";
 export { setTranslations } from "./utils/i18n";
 export { translate } from "./utils/i18n";
 export * from "./composites/Plugin/DashboardWidget";
-export { replacementVariablesShape, recommendedReplacementVariablesShape } from "./composites/Plugin/SnippetEditor/constants";
+export { replacementVariablesShape, recommendedReplacementVariablesShape } from "../search-metadata-previews/SnippetEditor/constants";
```
Comment (bintzandt) [inline packages/yoast-components/index.js]:
```suggestion
export { replacementVariablesShape, recommendedReplacementVariablesShape } from "@yoast/search-metadata-previews/SnippetEditor/constants";
```

File: packages/search-metadata-previews/SnippetPreview/components/HelpTextWrapper.js (hunk 4 above)
Comment (diedexx) [reply packages/search-metadata-previews/SnippetPreview/components/HelpTextWrapper.js]: Yup

File: packages/search-metadata-previews/jest/setupTests.js (hunk 6 above)
Comment (diedexx) [reply packages/search-metadata-previews/jest/setupTests.js]: haha I found the exact opposite. There are only one or two tests that require it

**Pull request merged**